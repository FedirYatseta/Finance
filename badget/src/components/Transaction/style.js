import styled from 'styled-components'


export const Wrapper = styled.div`
        display: flex;
        background: ${({ value }) => value < 0 ? '#FB0930' : '#0DDB2E'};
        border-radius: 4px;
        padding: 5px;
        margin: 5px 10px;
`;


export const Element = styled.div`
        background: ${({ value }) => value < 0 ? '#FB0930' : '#0DDB2E'};
        border-radius: 4px;
        padding: 5px;
        min-width: 150px;
`;