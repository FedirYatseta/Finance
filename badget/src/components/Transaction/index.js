import React from 'react'
import { Wrapper, Element } from './style'
import PropTypes from 'prop-types';

const Transaction = ({ transaction: { comment, date, value } }) => {

    return (
        <Wrapper value={value}>
            <Element> Value : {value}</Element>
            <Element> Date : {date}</Element>
            <Element> Comment : {comment}</Element>
        </Wrapper>
    )

}

Transaction.propTypes = {
    transaction: PropTypes.shape({
        value: PropTypes.number,
        date: PropTypes.string,
        comment: PropTypes.string
    })

}

export default Transaction