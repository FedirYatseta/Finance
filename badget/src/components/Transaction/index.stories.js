import React from 'react';

import Transaction from '.';

export default {
    title: 'Example/Transaction',
    component: Transaction,
};
const Template = (args) => <Transaction {...args}></Transaction>;

export const Income = Template.bind({});
Income.args = {
    transaction: {
        label: "some income",
        value: 23,
    }

};