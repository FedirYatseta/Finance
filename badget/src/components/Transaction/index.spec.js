

//import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import Transaction from '.';



describe('Transaction', () => {
  it('should show Transaction', () => {
    let comment = "test";
    let date = "2022-21-21";
    let value = 312.20;
    let transaction = { transaction: { comment, date, value } }
    const component = renderer.create(
      <Transaction transaction={transaction} />
    );
    expect(component).toMatchSnapshot();

  })
});