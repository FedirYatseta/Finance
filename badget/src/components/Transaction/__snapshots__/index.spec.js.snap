// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction should show Transaction 1`] = `
<div
  className="sc-bczRLJ cKPmbC"
>
  <div
    className="sc-gsnTZi irJlqJ"
  >
     Value : 
  </div>
  <div
    className="sc-gsnTZi irJlqJ"
  >
     Date : 
  </div>
  <div
    className="sc-gsnTZi irJlqJ"
  >
     Comment : 
  </div>
</div>
`;
